File: JavaScript/ProjectEuler0003.js
// projecteuler.net/problem=3

// Answer: 6857

// github.com/firatksee

function primeFactorsOf(number) {
    const primeFactors = new Set();
    let divisor = 2;

    while (number > 1) {
        if (number % divisor === 0) {
            primeFactors.add(divisor);
            number /= divisor;
        } else {
            divisor++;
        }
    }

    return Array.from(primeFactors);
}

const maxPrimeFactor = Math.max(...primeFactorsOf(600851475143));

console.log(maxPrimeFactor);

File: Python/ProjectEuler0003.py

# projecteuler.net/problem=3

# Answer: 6857

# github.com/firatksee


def primeFactorsOf(number):
    primeFactors, divisor = set(), 2

    while number > 1:
        if number % divisor == 0:
            primeFactors.add(divisor)
            number /= divisor
        else:
            divisor += 1
    return primeFactors


if __name__ == "__main__":
    print(max(primeFactorsOf(600851475143)))
